Source/Model/Reader/SecureContent101/NMR_ModelReaderNode_KeyStoreCEKParams.cpp

```diff
@@ -107,22 +107,30 @@ namespace NMR {
 		__NMRASSERT(pChildName);
 		__NMRASSERT(pXMLReader);
 		__NMRASSERT(pNameSpace);
-		if (strcmp(pNameSpace, XML_3MF_NAMESPACE_SECURECONTENTSPEC) == 0) {
-			if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_IV) == 0) {
-				PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
-				pXMLNode->parseXML(pXMLReader);
-				m_iv = base64_decode(pXMLNode->getValue());
-			} else if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_TAG) == 0) {
-				PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
-				pXMLNode->parseXML(pXMLReader);
-				m_tag = base64_decode(pXMLNode->getValue());
-			} else if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_AAD) == 0) {
-				PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
-				pXMLNode->parseXML(pXMLReader);
-				m_aad = base64_decode(pXMLNode->getValue());
-			} else {
-				m_pWarnings->addWarning(NMR_ERROR_NAMESPACE_INVALID_ELEMENT, mrwInvalidOptionalValue);
+		try{
+			if (strcmp(pNameSpace, XML_3MF_NAMESPACE_SECURECONTENTSPEC) == 0) {
+				if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_IV) == 0) {
+					PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
+					pXMLNode->parseXML(pXMLReader);
+					std::string result = base64_decode(pXMLNode->getValue());
+					m_iv = std::vector<nfByte>(result.begin(), result.end());
+				} else if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_TAG) == 0) {
+					PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
+					pXMLNode->parseXML(pXMLReader);
+					std::string result = base64_decode(pXMLNode->getValue());
+					m_tag = std::vector<nfByte>(result.begin(), result.end());
+				} else if (strcmp(pChildName, XML_3MF_SECURE_CONTENT_AAD) == 0) {
+					PModelReaderNode_StringValue pXMLNode = std::make_shared<CModelReaderNode_StringValue>(m_pWarnings);
+					pXMLNode->parseXML(pXMLReader);
+					std::string result = base64_decode(pXMLNode->getValue());
+					m_aad = std::vector<nfByte>(result.begin(), result.end());
+				} else {
+					m_pWarnings->addWarning(NMR_ERROR_NAMESPACE_INVALID_ELEMENT, mrwInvalidOptionalValue);
+				}
 			}
 		}
+		catch (...) {
+			// base64_decode throws an exception if the input is not valid base64
```

Comment:
please add specific error checking for base64_decode instead of this egeneric catch-all block.
As it is in this change, it will hide other exceptions in e.g. `pXMLNode->parseXML(pXMLReader);`

Reply:
okay its a runtime crash, should I introduce another Lib3mf error or use one which is already defined ?

Reply:
maybe you find a fitting one. If not, please create a new onw next to the other secure content ones.